packages/trading-widget/src/trading-widget/components/common/balance/withdraw-assets-composition-table.tsx

```diff
@@ -0,0 +1,80 @@
+import type { FC } from 'react'
+
+import { DEFAULT_VISIBLE_ASSETS_LIMIT } from 'core-kit/const'
+import type { useWithdrawAssetsInfo } from 'core-kit/hooks/trading/withdraw-v2/use-withdraw-assets-info'
+import {
+  formatNumberToLimitedDecimals,
+  formatToUsd,
+  formatUnits,
+} from 'core-kit/utils'
+import { TokenBadge } from 'trading-widget/components/common/index'
+import {
+  AssetsCompositionDisclosure,
+  AssetsCompositionTable,
+} from 'trading-widget/components/common/meta/assets-composition-disclosure/assets-composition-disclosure'
+import { useConfigContextParams } from 'trading-widget/providers/config-provider'
+
+interface WithdrawAssetsCompositionTableProps {
+  assets: ReturnType<typeof useWithdrawAssetsInfo>['data']
+}
+
+type AssetItem = NonNullable<
+  ReturnType<typeof useWithdrawAssetsInfo>['data']
+>[number]
+
+export const WithdrawAssetsCompositionTable: FC<
+  WithdrawAssetsCompositionTableProps
+> = ({ assets = [] }) => {
+  const { stablePrecision } = useConfigContextParams()
+
+  const visibleAssets = assets.slice(0, DEFAULT_VISIBLE_ASSETS_LIMIT)
+  const hiddenAssets = assets.slice(DEFAULT_VISIBLE_ASSETS_LIMIT)
+
+  const renderRow = ({
```

Comment:
Can it be moved out of component?

Reply:
wrapped in `useCallback` as it uses `stablePrecision` from the context